pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -185,17 +186,18 @@ func (p *criStatsProvider) ImageFsStats() (*statsapi.FsStats, error) {
 			UsedBytes:  &fs.UsedBytes.Value,
 			InodesUsed: &fs.InodesUsed.Value,
 		}
-		imageFsInfo := p.getFsInfo(fs.StorageId)
-		if imageFsInfo != nil {
-			// The image filesystem UUID is unknown to the local node or
-			// there's an error on retrieving the stats. In these cases, we
-			// omit those stats and return the best-effort partial result. See
-			// https://github.com/kubernetes/heapster/issues/1793.
-			s.AvailableBytes = &imageFsInfo.Available
-			s.CapacityBytes = &imageFsInfo.Capacity
-			s.InodesFree = imageFsInfo.InodesFree
-			s.Inodes = imageFsInfo.Inodes
+		imageFsInfo, err := p.getFsInfo(fs.StorageId)
+		if err != nil {
+			return nil, fmt.Errorf("imageFs information is unavailable: %v", err)
 		}
+		// The image filesystem UUID is unknown to the local node or
+		// there's an error on retrieving the stats. In these cases, we
+		// omit those stats and return the best-effort partial result. See
+		// https://github.com/kubernetes/heapster/issues/1793.
```

Comment:
This PR goes against the comment written here (that partial result should be returned).
What issue does returning an error fix?

/cc @yguo0905

Reply:
> What issue does returning an error fix?

@yujuhong Even if `imageFsInfo` is nil, the function will still `return s, nil`. This is not right. This will get panic when calling `s.AvailableBytes` since it has never been assigned a valid value.

We should float the error as well.

Reply:
The idea is that we want to provide as many metrics as possible when there are failures. Where did you get the panic? Is it possible to change the caller to check `s.AvailableBytes` before dereferencing it?